creusot/src/clone_map.rs

```diff
@@ -847,10 +847,12 @@ fn still_specializable<'tcx>(
         let is_final = tcx.impl_defaultness(def_id).is_final();
         matches!(impl_source.unwrap(), ImplSource::Param(_, _)) && !is_final
     } else if let Some(impl_id) = tcx.impl_of_method(def_id) && tcx.trait_id_of_impl(impl_id).is_some() {
-        let impl_source = resolve_impl_source_opt(tcx, param_env, def_id, substs).unwrap();
-
+        let impl_source = resolve_impl_source_opt(tcx, param_env, def_id, substs);
+        if impl_source.is_none() {
+            return true
+        }
```

Comment:
Do you want me to fill a bug report or are you handling that now?

Reply:
The issue seems to come from https://github.com/xldenis/creusot/blob/master/creusot-contracts/src/std/iter.rs#L80-L82

Reply:
Then when we define `Iterator` for `MapInv` we inherit the specification for `collect`, and get an invalid refinement since we need `IntoIterator` but the `extern_spec` wants us to use `Iterator` instead.